core/rawdb/accessors_chain.go

```diff
@@ -470,21 +477,35 @@ func WriteTransactions(db kv.RwTx, txs []types.Transaction, baseTxId uint64) err
 		}
 
 		// If next Append returns KeyExists error - it means you need to open transaction in App code before calling this func. Batch is also fine.
-		if err := db.Append(kv.EthTx, txIdKey, common.CopyBytes(buf.Bytes())); err != nil {
-			return err
+		if true {
```

Comment:
probably missed config

Reply:
yeah, still trying to figure out a way how to pass the config value here.

Reply:
you can convert `blockHash` to `*libcommon.Hash` and check if it is not `nil`. In the future the `else` branch will eventually be removed

Reply:
Some tests started to fail here. Need to think of some other way to check.

Reply:
This part introduced some test failures. Maybe I added the check incorrectly? Did you mean that, whenever `blockHash` is not nil the previous logic should work? @AlexeyAkhunov

Reply:
@nanevardanyan check itself is ok. Problem is: wrong data sent from outside of this function. In short: it's always not nil here (when transactionsV3 enabled and when disabled).

Reply:
okay if the check is correct, then it's true, and should I add the flag check and pass nil based on that?

Reply:
Any backward-compatible solution is ok for me